python/tvm/relay/qnn/op/qnn.py

```diff
@@ -113,7 +113,10 @@ def concatenate(data,
 
     # Find the dtype of the input expr. This is required for the requantize op. Since, this is
     # concatenate op, the dtype of the input is same as dtype of the output.
-    data0 = relay.transform.infer_type(data[0])
+    mod = relay.Module.from_expr(data[0])
+    mod = relay.transform.InferType()(mod)
+    entry = mod["main"]
+    data0 = entry if isinstance(data[0], relay.Function) else entry.body
```

Comment:
Why not use `from ...frontend.common import infer_type`. It seems unnecessary duplication of code. If the namespace is a problem we should probably move the existing method to a better place?

Reply:
Also, it seems the method `from ...frontend.common import infer_type` has more checks.

Reply:
It seems strange to import pass like `infer_type` from `frontend`. Should we consider create shortcut for `infer_type`? @jroesch

Reply:
Shortcut seems like a good idea.

Maybe, we can get this in to unblock CI and then follow up with another PR for shortcut and related changes to remove code duplication.